doc/manual/src/introduction.md

```diff
@@ -1,193 +1,144 @@
 
 ## Atomic upgrades and rollbacks
 
-Since package management operations never overwrite packages in the
-Nix store but just add new versions in different paths, they are
-_atomic_.  So during a package upgrade, there is no time window in
-which the package has some files from the old version and some files
-from the new version — which would be bad because a program might well
-crash if it’s started during that period.
+Since files in the Nix store are never changed, new versions are added to different paths, and environments are constructed from symlinks to these paths.
 
-And since packages aren’t overwritten, the old versions are still
-there after an upgrade.  This means that you can _roll back_ to the
-old version:
+Swapping symlinks to paths is _atomic_.
+With Nix, during a software upgrade, there is no time window in which a package has some files from the old version and some files from the new version.
+This eliminates inconsistencies by construction.
 
-```console
-$ nix-env --upgrade --attr nixpkgs.some-package
-$ nix-env --rollback
-```
+And since files aren’t overwritten, the old versions are still there after an upgrade.
+This means that you can instantly _roll back_ to an old version.
 
 ## Garbage collection
 
-When you uninstall a package like this…
+Files are’t deleted from the system right away — after all, you might want to do a rollback, or they might be in the profiles of other users.
+Instead, unused packages can be deleted safely by running the _garbage collector_.
 
-```console
-$ nix-env --uninstall firefox
-```
+This deletes all packages that aren’t in use by a currently running program or otherwise.
 
-the package isn’t deleted from the system right away (after all, you
-might want to do a rollback, or it might be in the profiles of other
-users).  Instead, unused packages can be deleted safely by running the
-_garbage collector_:
+## Transparent source/binary deployment
 
-```console
-$ nix-collect-garbage
-```
+Nix expressions generally describe how to build software from source, starting from some initial binary that is taken for granted.
+This is a _source deployment model_.
 
-This deletes all packages that aren’t in use by any user profile or by
-a currently running program.
+But since outputs are cached, building is only required if the output in question are not already in the Nix store.
+Nix can also automatically skip building from source and instead use a _binary cache_, a web server that provides pre-built binaries.
+This is a _binary deployment model_.
 
-## Functional package language
+[Hydra](https://hydra.nixos.org/), the Nix build cluster, has been producing binary artifacts and making them available in the [public cache](http://cache.nixos.org/) for many years.
+The cache allows running software going back as far as 2010, especially where sources cannot be obtained otherwise any more.
```

Comment:
Are not in process to thin out the history of the binary cache?

Reply:
That was written before thinning out was on the table. But keeping FODs should still preserve this property.